Gems/EMotionFX/Code/MCore/Source/Algorithms.cpp

```diff
@@ -367,8 +368,8 @@ namespace MCore
 
             // For quaternion, taking the mathematical summation of the quaternions and taking average doesn't yield the most accurate result.
             // However, it will yield close to the result when given quaternions within similar ranges.
-            size_t left = AZStd::min(i, sampleNum);
-            size_t right = AZStd::min(data.size() - 1 - i, sampleNum);
+            const size_t left = AZStd::min(i, sampleNum);
+            const size_t right = AZStd::min(data.size() - 1 - i, sampleNum);
             result += AZStd::accumulate(data.begin() + i - left, data.begin() + i + right + 1, AZ::Quaternion(0));
             result /= (float)(left + right + 1);
             result.Normalize();
```

Comment:
suggestion: Same comment about the cast as before (and I wonder if we could move this duplicated code to a single function that's templated on the type to accumulate... just seems a shame to have this very similar code in two places 🤔)

Reply:
Yeah I thought about that too. For rotation there is two problem, 1) is that we are operating on the compressedQuaternion and we have to deal with uncompress / compress the data, 2) is that there is a chance we need to revisit the algorithms because doing a mathematical average only works for rotation that changes slightly between frame - means that for most common cases it works, but some animation with crazy rotation it might causes weird result.
Personally I'd keep the rotation smoothing to be a separate function. But if we need to add more smoothing algorithm that doing a simple mathematical average we can switch to a template.

Reply:
Okay makes sense, thanks for the explanation :+1: